File: javascript/test/Structural/Composite.js
(function() {
  define(['Structural/Composite/TinOfTeaBags', 'Structural/Composite/OneTeaBag'], function(TinOfTeaBags, OneTeaBag) {
    'use strict';
    return describe('Test Structural Adapter', function() {
      return it('Creating tinOfTeaBags', function() {
        var smallTinOfTeaBags, teaBag1, teaBag2, teaBag3, tinOfTeaBags;
        tinOfTeaBags = new TinOfTeaBags("tin of tea bags");
        expect("The tinOfTeaBags has " + (tinOfTeaBags.countTeaBags()) + " tea bags in it.").toEqual("The tinOfTeaBags has 0 tea bags in it.");
        teaBag1 = new OneTeaBag("tea bag 1");
        expect("The teaBag1 has " + (teaBag1.countTeaBags()) + " tea bags in it.").toEqual("The teaBag1 has 1 tea bags in it.");
        teaBag2 = new OneTeaBag("tea bag 2");
        expect("The teaBag2 has " + (teaBag2.countTeaBags()) + " tea bags in it.").toEqual("The teaBag2 has 1 tea bags in it.");
        expect(tinOfTeaBags.add(teaBag1)).toBeTruthy();
        expect(tinOfTeaBags.add(teaBag2)).toBeTruthy();
        expect("The tinOfTeaBags now has " + (tinOfTeaBags.countTeaBags()) + " tea bags in it.").toEqual("The tinOfTeaBags now has 2 tea bags in it.");
        smallTinOfTeaBags = new TinOfTeaBags("small tin of tea bags");
        expect("The smallTinOfTeaBags has " + (smallTinOfTeaBags.countTeaBags()) + " tea bags in it.").toEqual("The smallTinOfTeaBags has 0 tea bags in it.");
        teaBag3 = new OneTeaBag("tea bag 3");
        expect(smallTinOfTeaBags.add(teaBag3)).toBeTruthy();
        expect("The smallTinOfTeaBags now has " + (smallTinOfTeaBags.countTeaBags()) + " tea bags in it.").toEqual("The smallTinOfTeaBags now has 1 tea bags in it.");
        expect(tinOfTeaBags.add(smallTinOfTeaBags)).toBeTruthy();
        expect("The tinOfTeaBags now has " + (tinOfTeaBags.countTeaBags()) + " tea bags in it.").toEqual("The tinOfTeaBags now has 3 tea bags in it.");
        expect(tinOfTeaBags.remove(teaBag2)).toBeTruthy();
        return expect("The tinOfTeaBags now has " + (tinOfTeaBags.countTeaBags()) + " tea bags in it.").toEqual("The tinOfTeaBags now has 2 tea bags in it.");
      });
    });
  });

}).call(this);

File: java/t/test/Structural/Composite.java
//TestTeaBagsComposite.java - testing the composite

package test.Structural;

import static org.junit.Assert.assertEquals;

import org.junit.Test;
import org.junit.Ignore;
import org.junit.runner.RunWith;
import org.junit.runners.JUnit4;

import Structural.Composite.*;



public class Composite 
{
	@Test
   public void main() 
	{
		
		TeaBags tinOfTeaBags = new TinOfTeaBags("tin of tea bags");

		TeaBags teaBag1 = new OneTeaBag("tea bag 1");
		TeaBags teaBag2 = new OneTeaBag("tea bag 2");

		org.junit.Assert.assertEquals("OneTeaBag count",
			"The teaBag2 has 1 tea bags in it.",
			"The teaBag2 has " + teaBag2.countTeaBags() + " tea bags in it."); 

		org.junit.Assert.assertEquals( "Putting teaBag1 in tinOfTeaBags", 
			true,
			tinOfTeaBags.add(teaBag1)
		);
		org.junit.Assert.assertEquals( "Putting teaBag2 in tinOfTeaBags", 
			true,
			tinOfTeaBags.add(teaBag2)
		);

		org.junit.Assert.assertEquals("TinOfTeaBags count after teabag1 and teabag2 adds", 
			("The tinOfTeaBags now has 2 tea bags in it."),
			("The tinOfTeaBags now has " + tinOfTeaBags.countTeaBags() + " tea bags in it.")
		);


		TeaBags smallTinOfTeaBags = new TinOfTeaBags("small tin of tea bags");
		org.junit.Assert.assertEquals(
			"The smallTinOfTeaBags has 0 tea bags in it.",
			("The smallTinOfTeaBags has " + smallTinOfTeaBags.countTeaBags() + " tea bags in it.")
		);

		TeaBags teaBag3 = new OneTeaBag("tea bag 3");
		org.junit.Assert.assertEquals("Putting teaBag3 in smallTinOfTeaBags", 
			true,
			smallTinOfTeaBags.add(teaBag3)
		);
		org.junit.Assert.assertEquals("Small tin count",
			"The smallTinOfTeaBags now has 1 tea bags in it.",
			"The smallTinOfTeaBags now has " + smallTinOfTeaBags.countTeaBags() + " tea bags in it.");

		org.junit.Assert.assertEquals(
			true,
			tinOfTeaBags.add(smallTinOfTeaBags)
		);

		org.junit.Assert.assertEquals( "TinOfTeaBags count before removal ", 
			"The tinOfTeaBags now has 3 tea bags in it.",
			"The tinOfTeaBags now has " + tinOfTeaBags.countTeaBags() + " tea bags in it."
		);

		org.junit.Assert.assertEquals("Removing teaBag2 from tinOfTeaBags", 
			true,
			tinOfTeaBags.remove(teaBag2)
		);

		org.junit.Assert.assertEquals("TinOfTeaBags count after removal.", 
			"The tinOfTeaBags now has 2 tea bags in it.",
			"The tinOfTeaBags now has " + tinOfTeaBags.countTeaBags() + " tea bags in it."
		);
   }
}  
